src/arbitrary/_internals/ArrayArbitrary.ts

```diff
@@ -56,6 +68,16 @@ export class ArrayArbitrary<T> extends NextArbitrary<T[]> {
     return vs;
   }
 
+  private safeGenerate(mrng: Random, biasFactorItems: number | undefined, targetSize: number): NextValue<T> {
```

Comment:
Would probably be more optimized to put the safe wrapper around `generateNItemsNoDuplicates` or `generateNItems` themselves. Less computations at each run 🤔